pkg/ruler/api.go

```diff
@@ -0,0 +1,260 @@
+package ruler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/weaveworks/common/user"
+
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+// RegisterRoutes registers the configs API HTTP routes with the provided Router.
+func (r *Ruler) RegisterRoutes(router *mux.Router) {
+	for _, route := range []struct {
+		name, method, path string
+		handler            http.HandlerFunc
+	}{
+		{"get_rules", "GET", "/api/prom/api/v1/rules", r.rules},
+		{"get_alerts", "GET", "/api/prom/api/v1/alerts", r.alerts},
+	} {
+		level.Debug(util.Logger).Log("msg", "ruler: registering route", "name", route.name, "method", route.method, "path", route.path)
+		router.Handle(route.path, route.handler).Methods(route.method).Name(route.name)
+	}
+}
+
+type errorType string
+
+const (
+	errorBadData  errorType = "bad_data"
+	errorInternal errorType = "internal"
```

Comment:
If we wanna keep full compatibility with Prometheus, in Prometheus it's:
`ErrServer      ErrorType = "server_error"`

Also, instead of redefining them here, we can directly use them from Prometheus (using its `ErrorType` instead of `errorType`).

Reply:
Prometheus doesn't export this type in it's API. However, I noticed this type is reimplemented in the `prometheus/client_golang` repo so I will snag it from there.